src/main/java/org/opensearch/knn/quantization/models/quantizationState/MultiBitScalarQuantizationState.java

```diff
@@ -73,12 +112,31 @@ public void writeTo(StreamOutput out) throws IOException {
      * @throws IOException if an I/O error occurs during deserialization.
      */
     public MultiBitScalarQuantizationState(StreamInput in) throws IOException {
-        int version = in.readVInt(); // Read the version
+        int version = in.readVInt();
         this.quantizationParams = new ScalarQuantizationParams(in, version);
-        int rows = in.readVInt(); // Read the number of rows
+
+        int rows = in.readVInt();
         this.thresholds = new float[rows][];
         for (int i = 0; i < rows; i++) {
-            this.thresholds[i] = in.readFloatArray(); // Read each row as a float array
+            this.thresholds[i] = in.readFloatArray();
+        }
+
+        if (Version.fromId(version).onOrAfter(Version.V_3_1_0)) {
+            FloatArrayWrapper[] wrappedBelowMeans = in.readOptionalArray(FloatArrayWrapper::new, FloatArrayWrapper[]::new);
```

Comment:
Are we using below or above means for random rotation? If not, could we remove and add them in adc change?